experimental/unitsconversion/src/measureunit.rs

```diff
@@ -63,8 +63,28 @@ impl MeasureUnit {
     /// NOTE:
     ///    if the unit id is found, the function will return (unit id, part without the unit id and without `-` at the beginning of the remaining part if it exists).
     ///    if the unit id is not found, the function will return None.
-    fn get_unit_id<'data>(part: &'data str, trie: &ZeroTrie<ZeroVec<'data, u8>>) -> Option<usize> {
-        trie.get(part.as_bytes())
+    fn get_unit_id<'data>(
+        part: &'data str,
+        identifier_split: &mut std::str::Split<'data, char>,
+        trie: &ZeroTrie<ZeroVec<'data, u8>>,
+    ) -> Option<usize> {
+        let mut part = part.to_string();
```

Comment:
Use a `Cow` to avoid allocation in the general case.

Reply:
Great, I will do that right away.

In order to add the test cases, could you reply to this comment:
https://github.com/unicode-org/icu4x/pull/4422/files#r1419262946

I need to be able to read the provider.

Reply:
To use the ZeroTrie cursor, this means that we need to change the algorithm in getting the power and the si prefix. 

Shall we do it in another PR and test the performance. who knows which implementation is better in performance ?

Reply:
You can't allocate memory in hot library code like this. We have clients who are sensitive to it. Also, we have a lot of experiential evidence that memory allocations are one of the biggest single contributors to slow code, so I have no doubt that ZeroTrieCursor will be faster.

Reply:
Okay, no problem. actually, I am implementing it right now.